app/templates/src/main/java/package/social/_SocialController.java

```diff
@@ -0,0 +1,44 @@
+package <%=packageName%>.social;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.social.connect.Connection;
+import org.springframework.social.connect.web.ProviderSignInUtils;
+import org.springframework.social.support.URIBuilder;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.CookieValue;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.view.RedirectView;
+
+import javax.inject.Inject;
+
+@Controller
+@RequestMapping("/social")
+public class SocialController {
+    private final Logger log = LoggerFactory.getLogger(SocialController.class);
+
+    @Inject
+    private SocialService socialService;
+
+    @Inject
+    private ProviderSignInUtils providerSignInUtils;
+
+    @RequestMapping(value = "/signup", method = RequestMethod.GET)
+    public RedirectView signUp(WebRequest webRequest, @CookieValue("NG_TRANSLATE_LANG_KEY") String langKey) {
+        try {
+            final Connection<?> connection = providerSignInUtils.getConnectionFromSession(webRequest);
```

Comment:
This is not the only place, you have lots of "final" keywords, they don't do anything and add more code -> can you remove them? So it's consistent with the rest of the code, which doesn't use them